crates/wdl-analysis/src/document.rs

```diff
@@ -804,8 +821,9 @@ impl Document {
     pub fn has_errors(&self) -> bool {
```

Comment:
I just made this change to get it compiling, but I'm not sure we actually want this as part of the `Document` API as it is now. I'd prefer this to instead `chain` and `filter` _and return the errors_ rather than a boolean. We should be reporting the problems encountered rather than detecting and failing without information.

Reply:
that makes sense to me, thanks for the extra context! I'll add that another thought I had while writing this is that we have this logic (filtering all the diagnostics to just the errors) in a bunch of places. It's trivial logic, but its led to a fair bit of repetition we could nip with a tweak in this API.